### ISSUE TITLE
dyld: Library not loaded

### ISSUE THREAD

**Issue by sts2055** [realm/realm-cocoa#1850]
My swift app runs fine in the simulator, but trying to run it on the device produces the following dyld`dyld_fatal_error:

```
dyld: Library not loaded: @rpath/RealmSwift.framework/Frameworks/Realm.framework/Realm
  Referenced from: /private/var/mobile/Containers/Bundle/Application/CE400CDC-B6C9-4610-B34B-F5B63F0ADC1E/Chirurgie Plastique FAQ.app/Frameworks/RealmSwift.framework/RealmSwift
  Reason: no suitable image found.  Did find:
	/private/var/mobile/Containers/Bundle/Application/CE400CDC-B6C9-4610-B34B-F5B63F0ADC1E/Chirurgie Plastique FAQ.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm: code signature invalid for '/private/var/mobile/Containers/Bundle/Application/CE400CDC-B6C9-4610-B34B-F5B63F0ADC1E/Chirurgie Plastique FAQ.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm'

	/private/var/mobile/Containers/Bundle/Application/CE400CDC-B6C9-4610-B34B-F5B63F0ADC1E/Chirurgie Plastique FAQ.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm: code signature invalid for '/private/var/mobile/Containers/Bundle/Application/CE400CDC-B6C9-4610-B34B-F5B63F0ADC1E/Chirurgie Plastique FAQ.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm'

	/private/var/mobile/Containers/Bundle/Application/CE400CDC-B6C9-4610-B34B-F5B63F0ADC1E/Chirurgie Plastique FAQ.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm: code signature invalid for '/private/var/mobile/Containers/Bundle/Application/CE400CDC-B6C9-4610-B34B-F5B63F0ADC1E/Chirurgie Plastique FAQ.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm'

	/private/var/mobile/Containers/Bundle/Application/CE400CDC-B6C9-4610-B34B-F5B63F0ADC1E/Chirurgie Plastique FAQ.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm: code signature invalid for '/private/var/mobile/Containers/Bundle/Application/CE400CDC-B6C9-4610-B34B-F5B63F0ADC1E/Chirurgie Plastique FAQ.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm'
```

Trying to run any of the examples that come with realm produces the same error (e.g. running  'Backlink'):
```
dyld: Library not loaded: @rpath/RealmSwift.framework/Frameworks/Realm.framework/Realm
  Referenced from: /private/var/mobile/Containers/Bundle/Application/36E0F42D-1D88-477E-8452-DDE20830A7A3/Backlink.app/Frameworks/RealmSwift.framework/RealmSwift
  Reason: no suitable image found.  Did find:
	/private/var/mobile/Containers/Bundle/Application/36E0F42D-1D88-477E-8452-DDE20830A7A3/Backlink.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm: code signature invalid for '/private/var/mobile/Containers/Bundle/Application/36E0F42D-1D88-477E-8452-DDE20830A7A3/Backlink.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm'

	/private/var/mobile/Containers/Bundle/Application/36E0F42D-1D88-477E-8452-DDE20830A7A3/Backlink.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm: code signature invalid for '/private/var/mobile/Containers/Bundle/Application/36E0F42D-1D88-477E-8452-DDE20830A7A3/Backlink.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm'
```

Since the error message states 'code signature invalid' I updated my signing identities, but that doesn't seem to be related as it doesn't fix the issue.
Been looking for a some time now, but haven't found a solution yet.

**Comment by segiddins** [comment]
@sts2055 what version of RealmSwift are you using? How did you install RealmSwift?

**Comment by sts2055** [comment]
I'm using realm-swift-0.92.0

**Comment by marcvanolmen** [comment]
I have same issue, I just followed the instructions by downloading:

https://github.com/realm/realm-cocoa/releases/download/v0.92.0/realm-swift-0.92.0.zip

Using the framework (I currently don't do step 5. because that causes an issue with Code Signed framework being modified.

**Comment by segiddins** [comment]
Did you make sure to do step (3), which is setting the Framework Search Paths to look for the embedded `Realm.framework` that's inside `RealmSwift.framework`?

**Comment by marcvanolmen** [comment]
Indeed i added the Extra path: /Frameworks, so in my case this looks like this:

$(PROJECT_DIR)/Cameo/Frameworks/RealmSwift.framework/Frameworks

**Comment by marcvanolmen** [comment]
made me think, step 3 is related to compiling this is an issue regarding running on the device.

**Comment by segiddins** [comment]
I couldn't reproduce this... Is there any change you could upload a sample project that demonstrates the issue?

**Comment by sts2055** [comment]
I was able to consistently address the issue by:

- replacing the script with the changes in this commit ff4477607a46df17f298610d61953f30be4405b9
- Deleting the app from the device
- build and run on the device
- the first time you run it on the device it will produce the same dyld`dyld_fatal_error outlined above
- the second time you run the app on the device the bug is gone

**Comment by marcvanolmen** [comment]
Maybe this could be the case: 

@segiddins I think you don't see this issue because you are signing your iOS apps with the same signatures as the one that is embedded in the RealmSwift.framework.

**Comment by jpsim** [comment]
@segiddins pushed a fix in #1855 and we'll be releasing 0.92.1 shortly. Thanks for your help in reporting these issues!
